### PR TITLE
fix test server/node read leader with unapplied log

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
When transfer 2 to leader, if 2 can't get HeartbeatResponse for a while,  check quorum will step down 2 to follower

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM.
PTAL @BusyJay

**Comment by zhangjinpeng1987** [comment]
@BusyJay PTAL

File: tests/raftstore/test_multi.rs [hunk 1]
```diff
@@ -410,6 +401,14 @@ fn test_read_leader_with_unapplied_log<T: Simulator>(cluster: &mut Cluster<T>) {
     // in this situation we need use raft read
     must_get_none(&cluster.get_engine(2), k);
 
+    // internal read will use raft read no matter read_quorum is false or true, cause applied
+    // index's term not equal leader's term, and will failed with timeout
+    if let Err(Error::Timeout(_)) = cluster.get_impl(k, false) {
```
Comment (BusyJay) [inline tests/raftstore/test_multi.rs]: Why not just use `call_command_on_leader`?

File: tests/raftstore/test_multi.rs [hunk 2]
```diff
@@ -410,6 +401,14 @@ fn test_read_leader_with_unapplied_log<T: Simulator>(cluster: &mut Cluster<T>) {
...
+        debug!("raft read failed with timeout");
+    } else {
+        assert!(false);
```
Comment (BusyJay) [inline tests/raftstore/test_multi.rs]: Use `panic` instead.

File: tests/raftstore/test_multi.rs (hunk 1 above)
Comment (zhangjinpeng1987) [inline tests/raftstore/test_multi.rs]: OK

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**